src/qt/walletmodel.cpp

```diff
@@ -202,29 +203,33 @@ void WalletModel::pollBalanceChanged()
         waitLonger = 0;
     }
 
+    // Avoid recomputing wallet balances unless a tx changed or
+    // BlockTip notification was received.
+    if (!fForceCheckBalanceChanged && m_cached_best_block_hash == getLastBlockProcessed()) return;
+
     // Get required locks upfront. This avoids the GUI from getting stuck on
     // periodical polls if the core is holding the locks for a longer time -
     // for example, during a wallet rescan.
-    TRY_LOCK(cs_main, lockMain);
-    if (!lockMain)
-        return;
-    TRY_LOCK(wallet->cs_wallet, lockWallet);
-    if (!lockWallet)
-        return;
+    int chainHeight = m_client_model->getLastBlockProcessedHeight();
+    const uint256& blockHash = m_client_model->getLastBlockProcessed();
```

Comment:
might be better to use the `WalletModel` wrappers, that null-check `m_client_model` ?

Reply:
it's done down the commits line.